### PR TITLE
feat(cli): Implement XML Loader 

### PR DESCRIPTION
**Pull request by partik03**
Issues: #288 

This PR adds an XML bucket loader with pull and push methods for handling localized XML files. A normalization utility converts XML to JSON for consistent test comparisons, preventing errors from formatting variations.

- New: xmlLoader to load/save XML.
- Testing: Enhanced with XML normalization for stable, format-agnostic assertions.
Ensures reliable XML handling and prevents future test fragility.

### CODE REVIEW

**Comment by partik03** [comment]
@maxprilutskiy kindly review this one

**Comment by partik03** [comment]
@maxprilutskiy I think we can merge this one, check it once

**Comment by partik03** [comment]
@maxprilutskiy kindly check it once also

File: packages/cli/src/loaders/xml.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+export default function createXmlLoader(): ILoader<string, Record<string, any>> {
+  return createLoader({
+    async pull(locale, input) {
+      let result: Record<string, any> = {};
+      
+      try {
+        const parsed = await parseStringPromise(input, { explicitArray: false, mergeAttrs:false, normalize:true, preserveChildrenOrder:true, normalizeTags: true, includeWhiteChars:true, trim: true });
+        result = parsed;
+      } catch (error) {
```
Comment (maxprilutskiy) [inline packages/cli/src/loaders/xml.ts:22]: please fix code styling here, must be 

```
{
  field1: value1,
  field2: value2,
  ....
```

File: packages/cli/src/loaders/xml.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        console.error("Failed to parse XML:", error);
+        result = {};
+      }
```
Comment (maxprilutskiy) [inline packages/cli/src/loaders/xml.ts:25]: Let it throw, if it's not possible to parse the xml - then the execution must stop

File: packages/cli/src/loaders/xml.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    
+    async push(locale, data) {
+      try {
+        const builder = new Builder({ headless: true });        
+        const xmlOutput = builder.buildObject(data);
+        const expectedOutput = normalizeXMLString(xmlOutput);
+        return expectedOutput;
+      } catch (error) {
+        console.error("Failed to build XML:", error);
```
Comment (maxprilutskiy) [inline packages/cli/src/loaders/xml.ts:37]: same here, if impossible to build xml - let it throw, exit early, and be handled at the topmost level in the cli

Comment (partik03) [reply packages/cli/src/loaders/xml.ts:37]: done

File: packages/cli/src/loaders/xml.ts (hunk 2 above)
Comment (partik03) [reply packages/cli/src/loaders/xml.ts:25]: done

File: packages/cli/src/loaders/xml.ts (hunk 1 above)
Comment (partik03) [reply packages/cli/src/loaders/xml.ts:22]: done

**Review by maxprilutskiy — APPROVED** [review]
awesome job! 🔥

**Pull request merged**